presto-native-execution/presto_cpp/main/tests/TaskManagerTest.cpp

```diff
@@ -1253,42 +1255,94 @@ TEST_F(TaskManagerTest, getResultsFromFailedTask) {
   ASSERT_EQ(results->data->capacity(), 0);
 }
 
-TEST_F(TaskManagerTest, testCumulativeMemory) {
-  auto filePaths = makeFilePaths(10);
-  auto vectors = makeVectors(10, 1'000);
-  for (int i = 0; i < 10; ++i) {
+DEBUG_ONLY_TEST_F(TaskManagerTest, testCumulativeMemory) {
```

Comment:
It took me quite some time to figure out what was happening and how #22328 is fixing the problem. It feels like the code in PrestoTask is rather fragile and can break easily. This test feels quite heavy weight. It uses TestValue, which makes it debug-only, and performs a lot of moves that are hard to understand. Can we do something to simplify the code to make it less error prone and more easily testable?

https://github.com/prestodb/presto/pull/22328/files#r1540631632

Reply:
Simplify the test per our offline discussion. Thanks!